app/src/main/java/a3/com/convo/fragments/ProfileDetailsFragment.java

```diff
@@ -66,14 +72,38 @@ public void onClick(View view) {
             return;
         }
         tvUserName.setText(userName);
-        String userHometown = user.getString(Constants.HOMETOWN);
-        if (userHometown == null) {
+        String userHometownObjectId = user.getString(Constants.HOMETOWN);
+        if (userHometownObjectId == null) {
             Log.e("ProfileDetailsFragment", "User doesn't have hometown and that's fine");
             tvUserHometown.setText(Constants.EMPTY_STRING);
         }
         else {
-            tvUserHometown.setText(userHometown);
+            ParseQuery<Page> query = ParseQuery.getQuery(Page.class);
+            query.getInBackground(userHometownObjectId, new GetCallback<Page>() {
+                @Override
+                public void done(Page object, ParseException e) {
+                    if (e == null && object != null) {
+                        String hometown_name = object.getName();
+                        if (hometown_name == null || hometown_name.isEmpty()){
+                            Log.e("ProfileDetailsFragment", "User's hometown object has no name");
+                            return;
+                        }
+                        tvUserHometown.setText("From " + hometown_name);
+                    }
+                    else {
+                        Log.e("hometown", "hometown not showing up");
+                        e.printStackTrace();
+                    }
+                }
+            });
         }
+        Integer gamesPlayed = ((Integer)user.getNumber(Constants.NUM_GAMES));
```

Comment:
Check if user is instanceof Integer before casting it to Integer to avoid ClassCastException